src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -82,15 +82,20 @@ class FastSync(
   }
 
   def waitingForTargetBlock: Receive = handleCommonMessages orElse {
-    case FastSyncTargetBlockSelector.Result(targetBlockHeader) =>
+    case FastSyncTargetBlockSelector.Result(targetBlockHeader, mostUpToDateBlockHeader) =>
       if (targetBlockHeader.number < 1) {
         log.info("Unable to start block synchronization in fast mode: target block is less than 1")
         appStateStorage.fastSyncDone()
         context become idle
         syncController ! Done
       } else {
         val initialSyncState =
-          SyncState(targetBlockHeader, safeDownloadTarget = targetBlockHeader.number + syncConfig.fastSyncBlockValidationX)
+          SyncState(
```

Comment:
I would move it to a method in `SyncState` companion object

Reply:
Why would you want to move it? It is a simple constructor call